pkgs/top-level/python-packages.nix

```diff
@@ -14204,7 +14204,9 @@ self: super: with self; {
 
   scikit-tda = callPackage ../development/python-modules/scikit-tda { };
 
-  scipy = callPackage ../development/python-modules/scipy { };
+  scipy = callPackage ../development/python-modules/scipy {
+    numpy = numpy_2;
```

Comment:
We can't do that within `python3Packages`.

Reply:
> We can't do that within `python3Packages`.

Why?

Reply:
That's a semantic change - I won't mind using `numpy_2` directly in `scipy/default.nix`.

Reply:
> I won't mind using `numpy_2` directly in `scipy/default.nix`.

That's not the issue.
Your change breaks `python3.withPackages (ps: with ps; [ numpy scipy ])`.

Reply:
Never mind, that's not true.
But it does break `python3.withPackages (ps: with ps; [ scipy ])`.

Reply:
> Never mind, that's not true. But it does break `python3.withPackages (ps: with ps; [ scipy ])`.

OK so what is your opinion on the matter? Use `numpy = numpy_2` in `python-packages.nix`, or use `numpy_2` directly in the `scipy/default.nix` expression?

Reply:
Use `numpy` everywhere. And propagate it.